README.md

```diff
@@ -14,7 +14,72 @@ For more details, refer to the official [Camunda Reference Architecture document
 
 ## Structure
 
-The repository is structured into cloud providers (`aws`, `azure`, `general`) and internal-only reusable modules (`modules`).
+The repository is organized into different cloud providers (`aws`, `azure`, `generic`) and internal reusable modules (`modules`) that are associated with each cloud provider.
+
+### Naming Convention
+
+The directory structure follows a standardized naming pattern:
+
+```
+- {cloud_provider}
+  - modules
+  - {category}
+    - {solution}-{declination}-{feature}
+```
+
+Where:
+- `{cloud_provider}`: The cloud provider (`aws`, `azure`, `generic`).
+- `{category}`: The type of service or technology (e.g., `kubernetes`, `compute`).
+- `{solution}`: The specific solution, such as `eks` (Amazon EKS), `gke` (Google Kubernetes Engine), or `ec2` (Amazon EC2).
+- `{declination}`: A specific feature or deployment model, particularly in relation to **Camunda 8**, such as:
+  - `single-region` (deployment in a single region).
+  - `dual-region` (high availability across two regions).
+- `{feature}`: A variation of the solution, such as:
+  - `irsa` (native IAM auth of AWS).
+  - `spot-instances` (for EC2 cost optimization).
+
+### Modules
+
+The `modules` directory is tied to specific cloud providers. Each cloud provider may include reusable modules that can be utilized across multiple solutions within that cloud environment.
+
+### Example Structure
+
+For AWS Kubernetes and EC2 solutions:
+
+```
+- aws
+  - kubernetes
+    - eks-dual-region-spot-instances
+    - eks-single-region-on-demand
+  - compute
+    - ec2-single-region-on-demand
```

Comment:
I think we said that `on-demand` would be the default option and not mention it on top?